frontend/cypress/integration/navbar.spec.js

```diff
@@ -0,0 +1,79 @@
+import faker from 'faker'
+
+describe('Validates `Add Project` behavior', () => {
+  const username = faker.name.firstName()
+  const email = faker.internet.email()
+  const password = '123456'
+
+  before(() => {
+    cy.visit('/')
+    cy.createUser(username, email, password)
+  })
+
+  beforeEach(() => {
+    cy.signOut()
+  })
+
+  it('unauthenticated user', () => {
+    // The user is unauthenticated
+    cy.window().its('session.user').should('eq', null)
+
+    // Clicking `Add Project` redirects to the login page.
+    cy.get('#add_project').click()
+    cy.url().should('eq', 'http://kitspace.test:3000/login')
+  })
+
+  it('authenticated user', () => {
+    // sign the user in.
+    cy.visit('/login')
+    cy.stubSignInReq(true, { LoggedInSuccessfully: true })
+    cy.signIn(username, password)
+
+    // Clicking `Add Project` redirects to new project page.
+    cy.get('#add_project').click()
+    cy.url().should('eq', 'http://kitspace.test:3000/projects/new')
+  })
```

Comment:
Why is an authenticated user signing back in? I don't think we should allow that behaviour, rather say to the user: you are already signed in.

Reply:
To test the behavior for authenticated users, it needs an authenticated user. So it creates a random user and sign him in and proceed to check the actual logic. It's more of a setup step, not the test itself.